aten/src/THC/generic/THCTensor.cpp

```diff
@@ -212,7 +211,21 @@ THCTensor *THCTensor_(newView)(THCState *state, THCTensor *tensor, at::IntArrayR
     "not compatible with input tensor's size and stride (at least one dimension spans "
     "across two contiguous subspaces). Call .contiguous() before .view().");
   auto stride_value = *stride;
+
+  // NOTE: This path of constructing the Tensor directly with the viewed Storage is necessary
+  // to allow `view` not to have a device_guard.  Taking the common TH path of allocating a storage
+  // on the current device [via THCTensor_(new)] and then swapping out the storage later can change
+  // the device out from under the tensor.  Having the device be consistent through a Tensor's lifetime
+  // is an invariant we wish to keep to support caching, simplicity, etc.
+  auto storage = THStorage_wrap(tensor->storage().unsafeGetStorageImpl());
+  THCTensor *self = c10::make_intrusive<at::TensorImpl, at::UndefinedTensorImpl>(
```

Comment:
why not just `new`? :)

Reply:
That's the API :).  The `intrusive_ptr(TTarget*)` constructor is private, the API is to call `make_intrusive`.  Since the TH API is in terms of raw pointers we immediately `release` it and at the TH/ATen boundary we `reclaim` it.

More justification(s) here:
https://github.com/pytorch/pytorch/blob/486fae563d79989b31e5708fb488a1376aec391b/c10/util/intrusive_ptr.h#L199-L202
https://github.com/pytorch/pytorch/blob/486fae563d79989b31e5708fb488a1376aec391b/c10/util/intrusive_ptr.h#L18-L33